reposerver/repository/repository.go

```diff
@@ -486,8 +486,12 @@ func findManifests(appPath string, env *v1alpha1.Env, directory v1alpha1.Applica
 			objs = append(objs, &obj)
 		} else if strings.HasSuffix(f.Name(), ".jsonnet") {
 			vm := makeJsonnetVm(directory.Jsonnet, env)
+			jpaths := []string{appPath}
+			for _, p := range directory.Jsonnet.JPaths {
+				jpaths = append(jpaths, filepath.Join(appPath, p))
```

Comment:
We should really make sure that we don't jump out of `appPath` here, i.e. when `p` is `../../foo/bar` or similar. Paths should only be added if they are within `appPath` after calling `filepath.Join()`

Reply:
my usecase  like below:

```
├── environments
│   └── default
│       ├── app.yaml
│       ├── main.jsonnet
│       └── spec.json
├── jsonnetfile.json
├── jsonnetfile.lock.json
├── lib
└── vendor
    ├── k.libsonnet
    ├── k8s.libsonnet
    └── ksonnet-util
        ├── jaeger.libsonnet
        └── kausal.libsonnet
```

If `appPath` is `environments/default`  or something else like  `environments/XXX`, jpaths can be that `../../vendor` and `../../lib` .  So I need to `appRoot` (e.g.:  gitRoot or flocal file path of `spec.source.repoURL`?).  
IMHO, Paths shoule be there in `appRoot`. Is it right? Do you have a good idea about it? :->